speak/face.py

```diff
@@ -152,20 +144,35 @@ def update(self, status=None):
             self._eyebox.pack_start(the, True, True, FACE_PAD)
             the.show()
 
-        self._mouth = status.mouth(self._audio, self.fill_color)
+        self._mouth = status.mouth(self.fill_color)
         self._mouth.show()
         self._mouthbox.add(self._mouth)
 
     def say(self, something):
-        self._audio.speak(self._peding or self.status, something)
+        if self._pending is None:
+            voice_name = self.status.voice.name
+        else:
+            voice_name = self._pending.voice.name
+        all_voices = self.speech.get_all_voices()
+        self.say_text(something, voice_name, all_voices)
 
     def say_notification(self, something):
-        status = (self._peding or self.status).clone()
+        status = (self._pending or self.status).clone()
         status.voice = voice.defaultVoice()
-        self._audio.speak(status, something)
+        voice_name = status.voice.name
+        all_voices = self.speech.get_all_voices()
+        self.say_text(something, voice_name, all_voices)
+
+    def say_text(self, something, voice_name, all_voices):
```

Comment:
Because I pointed out the similarity of the code leading up to two calls to say_text.  This is a variant of say_text that uses a voice name not a language code.  To avoid the interpretation that this is a redefinition, it could be named say_text_by_voice_name, or my_say_text, or just _say.  say_notification could change to say_text_in_default_language; as it took me a while to figure out why it was called notification.